src/blob_storage.cc

```diff
@@ -42,16 +42,19 @@ Status BlobStorage::GetBlobFilesInRanges(const RangePtr* ranges, size_t n,
          it != ((end != nullptr) ? blob_ranges_.upper_bound(*end)
                                  : blob_ranges_.end());
          it++) {
-      // Obsolete files are to be deleted, so just skip.
-      if (it->second->is_obsolete()) continue;
+      // The file is obsolete or being GCed, so just skip.
+      if (it->second->file_state() != BlobFileMeta::FileState::kNormal) {
```

Comment:
is it better to explicitly (it->second->is_obsolete() ||  specify  == kBeingGC)  instead of a larger scope != BlobFileMeta::FileState::kNormal

Reply:
The larger scope makes it safer, hard to reason about whether other states would meet other corner cases